internal/services/contracts_events_consumer.go

```diff
@@ -101,18 +103,27 @@ func (c *ContractsEventsConsumer) processEvent(event *shared.CloudEvent[json.Raw
 		return err
 	}
 
-	switch data.EventName {
-	case PrivilegeSet.String():
-		c.log.Info().Str("event", data.EventName).Msg("Event received")
-		return c.setPrivilegeHandler(&data)
-	case AftermarketDeviceNodeMinted.String():
-		if data.Contract == c.registryAddr {
+	if event.Source == fmt.Sprintf("chain/%d", c.settings.DIMORegistryChainID) {
```

Comment:
Could we do check-and-return for this exceptional case? I think it would read better.